src/cli.rs

```diff
@@ -1017,4 +1046,102 @@ FLAGS:
             structopt::clap::ErrorKind::ArgumentConflict
         );
     }
+
+    //////////////////////////////////////////////////////////////////////////
+    // Directory structure tests, to be run with `--test-threads=1` to prevent
+    // ENV clobbering.
+    //////////////////////////////////////////////////////////////////////////
```

Comment:
It might be easier to test this stuff in `tests/lib.rs` where we execute the Sheldon binary as a subprocess so we can easily control its environment.

Reply:
It would be more laborious but less brittle, I agree.

I don't see a ready way to check in-memory `Settings` from integration tests, however. Shall I look into introducing one, or use the existing TestCommand/TestCase functionality to directly check that files are created within the expected folder structure?

Reply:
Hmmm, yeah looks like a bunch of work. I'm also okay with you leaving the tests like this and opening an issue for this and it can be done at a different time as well 👍. It will probably be easier to change this if `Settings` was updated to use `config_dir` and `data_dir`.

Reply:
As it happens, most of the necessary work to adapt integration tests to different directory structures is already done, which should speed up the deprecation of `root` nicely. I'll review the changes tonight so that we can merge this PR and move onto the next.